aten/src/ATen/core/dispatch/OperatorEntry.cpp

```diff
@@ -34,12 +35,11 @@ OperatorEntry::OperatorEntry(OperatorName&& operator_name)
 }
 
 namespace {
-  void checkSchema(const OperatorName& name, const FunctionSchema& from_def, const std::string& from_def_debug, const FunctionSchema& inferred, const std::string& inferred_debug) {
+  void checkSchema(const OperatorName& name, const FunctionSchema& from_def_, const std::string& from_def_debug, const KernelFunction& kernel, const FunctionSchema& inferred_, const std::string& inferred_debug) {
     // TODO: figure out if we can just directly save real schema at def time
-    c10::optional<std::string> schema_difference = findSchemaDifferences(
-      from_def.cloneWithRealTypes(),
-      inferred.cloneWithRealTypes()
-    );
+    FunctionSchema from_def = from_def_.cloneWithRealTypes(kernel.isValidSymUnboxed());
+    FunctionSchema inferred = inferred_.cloneWithRealTypes();
```

Comment:
so, if the C++ function decl had SymInt, the yaml must also be updated

if the yaml had SymInt, boxed or Unboxed-nonsymint kernels are allowed to persist for BC.  (and presumably somewhere there is codegen to convert the symint to int64)

ok this part looks right to me

Reply:
Yes.